src/matches.rs

```diff
@@ -59,6 +62,16 @@ impl LateLintPass for MatchPass {
                          expression to match: `match *{} {{ ...`", snippet(cx, ex.span, "..")));
                 }
             }
+
+            // check preconditions for MATCH_BOOL
+            // type of expression == bool
+            if is_bool_expr(cx, ex) {
+                if in_external_macro(cx, expr.span) { return; }
```

Comment:
Not sure if I understood this. Can you explain this a bit. Tests are not order dependent, right?